src/OrchardCore/OrchardCore.Navigation.Core/NavigationManager.cs

```diff
@@ -184,7 +169,15 @@ private List<MenuItem> ComputeHref(List<MenuItem> menuItems, ActionContext actio
         {
             foreach (var menuItem in menuItems)
             {
-                menuItem.Href = GetUrl(menuItem.Url, menuItem.RouteValues, actionContext);
+                if (menuItem.ReturnUrlRouteValues != null)
+                {
+                    var returnUrl = GetUrl(menuItem.Url, menuItem.ReturnUrlRouteValues, actionContext);
```

Comment:
Only thing remaining to look is what would happen if there was a menuItem.Url passed here and should it then always be null since we always pass route params for composing a returnUrl.

Reply:
removed code no issue